osu.Game/Overlays/MusicController.cs

```diff
@@ -419,15 +419,20 @@ public bool AllowRateAdjustments
         }
 
         /// <summary>
-        /// Resets the speed adjustments currently applied on <see cref="CurrentTrack"/> and applies the mod adjustments if <see cref="AllowRateAdjustments"/> is <c>true</c>.
+        /// Resets the track adjustments currently applied on <see cref="CurrentTrack"/> and applies the mod adjustments if <see cref="AllowRateAdjustments"/> is <c>true</c>.
         /// </summary>
         /// <remarks>
-        /// Does not reset speed adjustments applied directly to the beatmap track.
+        /// Does not reset any adjustments applied directly to the beatmap track.
         /// </remarks>
         public void ResetTrackAdjustments()
         {
             CurrentTrack.ResetSpeedAdjustments();
 
+            // Adjustments other than speed should also be reset
+            // e.g. ModMuted may apply volume adjustments
+            CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Balance);
+            CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Volume);
```

Comment:
This method is a hot mess now. First it does `ResetSpeedAdjustments()` (quite weirdly specific API on the drawable track's behalf), then the controller *manually* undoes the *other* adjustments, and then `allowRateAdjustments` is used to steer whether mod adjustments should be added back. Which at the point in which the "mute" mod exists are *no longer solely rate adjustments*.

Reply:
In addition, changes like this should be done in their own PR. Mod PRs should not modify any file other than the mod's class.

Reply:
Will open a new PR with more comprehensive changes when I have the time to. Just to confirm, is this an acceptable way to resolve the issue? (i.e. to change `ResetTrackAdjustments` and `AllowRateAdjustments` so that they account for all types of adjustments, not just speed)

Reply:
I'm not sure whether it's valid or not. Based on @bdach's response above, that is questionable. I'd rather have it in its own PR before continuing discussion, regardless.

Reply:
At the *very* least `allowRateAdjustments` needs to not be named that anymore. But as stated I also see no reason to not reset all adjustments in `DrawableTrack` rather than half-there half-in-music-controller. Unless there is one.